source/material_model/equation_of_state/thermodynamic_table_lookup.cc

```diff
@@ -0,0 +1,611 @@
+      }
+
+
+
+      template <int dim>
+      void
+      ThermodynamicTableLookup<dim>::
+      evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+               MaterialModel::MaterialModelOutputs<dim> &out,
+               std::vector<std::vector<double>> &mass_fractions,
+               std::vector<std::vector<double>> &volume_fractions) const
+      {
+        EquationOfStateOutputs<dim> eos_outputs (number_of_lookups());
+
+        fill_mass_and_volume_fractions (in, mass_fractions, volume_fractions);
+
+        for (unsigned int i=0; i < in.n_evaluation_points(); ++i)
+          {
+            // Evaluate the equation of state properties at the current evaluation point
+            evaluate_at_single_point(in, i, eos_outputs);
+
+            // The density and isothermal compressibility are both volume-averaged
+            out.densities[i] = MaterialUtilities::average_value (volume_fractions[i], eos_outputs.densities, MaterialUtilities::arithmetic);
+            out.compressibilities[i] = MaterialUtilities::average_value (volume_fractions[i], eos_outputs.compressibilities, MaterialUtilities::arithmetic);
+            out.entropy_derivative_pressure[i] = MaterialUtilities::average_value (mass_fractions[i], eos_outputs.entropy_derivative_pressure, MaterialUtilities::arithmetic);
+            out.entropy_derivative_temperature[i] = MaterialUtilities::average_value (mass_fractions[i], eos_outputs.entropy_derivative_temperature, MaterialUtilities::arithmetic);
+
+            // Only calculate the averaged non-reactive specific heat and
+            // thermal expansivity if latent heat is to be ignored.
+            if (!latent_heat)
+              {
+                // Specific heat is measured per unit mass, so it is mass averaged.
+                // Thermal expansivity is volume averaged.
+                out.specific_heat[i] = MaterialUtilities::average_value (mass_fractions[i], eos_outputs.specific_heat_capacities, MaterialUtilities::arithmetic);
+                out.thermal_expansion_coefficients[i] = MaterialUtilities::average_value (volume_fractions[i], eos_outputs.thermal_expansion_coefficients, MaterialUtilities::arithmetic);
+              }
+          }
+
+        // Calculate the specific heat and thermal expansivity
+        // including the effects of reaction.
+        if (latent_heat)
+          evaluate_thermal_enthalpy_derivatives(in, out);
+
+        // fill seismic velocity outputs if they exist
+        if (SeismicAdditionalOutputs<dim> *seismic_out = out.template get_additional_output<SeismicAdditionalOutputs<dim> >())
+          fill_seismic_velocities(in, out.densities, volume_fractions, seismic_out);
+
+        // fill phase volume outputs if they exist
+        if (NamedAdditionalMaterialOutputs<dim> *phase_volume_fractions_out = out.template get_additional_output<NamedAdditionalMaterialOutputs<dim> >())
+          fill_phase_volume_fractions(in, volume_fractions, phase_volume_fractions_out);
```

Comment:
I just noticed that you can't simply move those lines into the material model, since you create the additional outputs in the equation of state. But you could write a new function as part of the equation of state and then call that from the material model at the end of the evaluate function as `equation_of_state.fill_additional_outputs(in, volume_fractions, out)` or something like that. That should work.